main/org.thingpedia.iot.light-bulb/dataset.tt

```diff
@@ -61,15 +61,58 @@ dataset @org.thingpedia.iot.light-bulb language "en" {
     #[id=52298582]
     #[name="SetPowerOnLightBulbWithPower"];
 
+    action (p_color :Enum(white,black,red,orange,yellow,green,blue,purple,pink,brown,gray,chartreuse,cyan,azure,violet,magenta,rose,amber,teal,indigo)) = @org.thingpedia.iot.light-bulb.set_color(color=p_color)
```

Comment:
This works for now, but there are so many possible colors, I've been thinking we should make colors one of the builtin entities in ThingTalk, maybe using the x11/svg color list, or even using hex codes and some code in genie to do the mapping.

Reply:
Yes, agreed, actually that is an easy list common in iot (supported by many bulbs), hue has its own list of available color, also by name (every iot bulb that interact with voice assistant has the same problem). 

But obviously if we can have a background mapping to whatever color named by name, to the corrispondent rgb or hex will be better. Would be nice if I can ask for a "light blu" and then say "add a little bit more of yellow to it", or "less green", but then I should also be able to save it some how or otherwise ask for a color and have a parsed result could end up to be a little bit random

Reply:
Do the bulbs only do color by name? I thought they supported arbitrary RGB (subject to some color profile mapping)?
"Add a bit more yellow" is tricky though, unless we make Color a builtin type in ThingTalk with three RGB fields. Which maybe it's worth doing, if color is used in other skills.

Reply:
several different colors option available: https://www.home-assistant.io/integrations/light/, 

Could be possible to manage single color, but on any command the whole color should be issued because it overwrite the previous, so to say "more yellow" you need to know from which color you start and then add to that your "yellow amount", otherwise it will send a "only yellow" light. 
Unfortunately HA doesn't offer color state, and it should be saved somewhere by the user at the time it is issued to HA.